promise-types/http/http_promise_type.py

```diff
@@ -127,29 +151,21 @@ def evaluate_promise(self, promiser, attributes):
                 SSL_context = ssl.SSLContext()
                 SSL_context.verify_method = ssl.CERT_NONE
 
-        try:
-            if target:
-                # TODO: create directories
-                with open(target, "wb") as target_file:
-                    with urllib.request.urlopen(request, context=SSL_context) as url_req:
-                        if not (200 <= url_req.status <= 300):
-                            self.log_error("Request for '%s' failed with code %d" % (url, url_req.status))
-                            return (Result.NOT_KEPT, ["%s_%s_request_failed" % (canonical_promiser, method)])
-                        # TODO: log progress when url_req.headers["Content-length"] > REPORTING_THRESHOLD
-                        done = False
-                        while not done:
-                            data = url_req.read(512 * 1024)
-                            target_file.write(data)
-                            done = bool(data)
-            else:
-                with urllib.request.urlopen(request, context=SSL_context) as url_req:
-                    if not (200 <= url_req.status <= 300):
-                        self.log_error("Request for '%s' failed with code %d" % (url, url_req.status))
-                        return (Result.NOT_KEPT, ["%s_%s_request_failed" % (canonical_promiser, method)])
-                    done = False
-                    while not done:
+        try:      
+            with urllib.request.urlopen(request, context=SSL_context) as url_req:
+                if not (200 <= url_req.status <= 300):
```

Comment:
Interestingly, 300 code (rarely used), means "multiple choices" for redirect so it seems this should be `< 300` instead of `<= 300`.
```suggestion
                if not (200 <= url_req.status < 300):
```

Reply:
Added in separate commit.